src/Rules/Compile.hs

```diff
@@ -22,6 +29,7 @@ compilePackage rs context@Context {..} = do
         if ("//*.c" ?== src)
         then do
             need $ src : deps
+            mapM_  (needGenerated context) . filter ("//*.c" ?==) $ src : deps
```

Comment:
Another TODO: `TODO: Improve parallelism by collecting all dependencies and need'ing them all at once.`